### ISSUE TITLE
[CEP-34] Bug: Sorting Childoptions not refreshing cache for Cascading Field

### ISSUE THREAD

**Issue by drdownload** [codeclou/customfield-editor-plugin#34]
Hi, we try to update the sequence by using the sort option after we changed a childoption value.

This works on the option level but not for childoptions.

If we query the API afterwards everything looks fine, however the changed sequence is not reflected in JIRA.

**Comment by clouless** [comment]
Are you using the REST API or the Web-GUI?
Which plugin version do you use and what JIRA Version do you use?

I will try to investigate further.

**Comment by clouless** [comment]
this will be fixed in v1.5.0

**Comment by drdownload** [comment]
Wow dude, you are fast!

Unfortunately we are already using the REST-API (Plugin-Version: 1.3.0, JIRA-Version: 7.3.3)

The issue in more detail:
we have a cascading-select field that looks like this (we use a non-default context)

Parent1
	- Child1
	- Child2

Initial State:

GET user/customfields/10642/contexts/11162/options/10471/childoptions
```

[
    {
        "optionvalue": "Child1",
        "id": 10472,
        "sequence": 0,
        "disabled": false
    },
    {
        "optionvalue": "Child2",
        "id": 10473,
        "sequence": 1,
        "disabled": false
    }
]
```

Now we change the optionvalue of "Child1" to "zChild1" via the REST-API.

Then we call "PUT user/customfields/10642/contexts/11162/options/10471/childoptions/sort" which returns 200

Calling "GET user/customfields/10642/contexts/11162/options/10471/childoptions" again now shows the correct order
```

[
    {
        "optionvalue": "Child2",
        "id": 10473,
        "sequence": 0,
        "disabled": false
    },
    {
        "optionvalue": "zChild1",
        "id": 10472,
        "sequence": 1,
        "disabled": false
    }
]
```

However, in JIRA itself, they are still in the wrong order (zChild1 before Child2). Both in the JIRA adminsection where you can manually edit the custom field aswell as on the issue screens.
As soon as we click "Sort options alphabetically" on the childoptions-edit-screen in the JIRA admin section, the options get sorted correctly.
"GET user/customfields/10642/contexts/11162/options/10471/childoptions" shows no difference before and after clicking "Sort options aphabetically"

We want to only use the REST-API without anyone having to click, since we are building a bigger sync-job that manages parent- and child-options by using another system as source.

Thank you for your help and fast reply!

**Comment by clouless** [comment]
I already fixed it for v1.5.0 but you will ned JIRA 7.5.0 since I only fix bugs and implement features in a roll-forward strategy. I am planning to release v1.5.0 in around 1-2 weeks.
So you might need to upgrade JIRA.

**Comment by drdownload** [comment]
Thx a lot. Great support!

**Comment by clouless** [comment]
Also great bug report :)
Thx for pointing that out :)

**Comment by clouless** [comment]
released with 1.5.0 on 2017-10-07
